tests/apps/coverage/test_coverage.py

```diff
@@ -0,0 +1,183 @@
+    group_id = "group_id"
+    group_name = "group_name"
+    bed_file = "bed_file"
+    mocked_run_command = mocker.patch.object(Process, "run_command")
+    api = ChanjoAPI(CHANJO_CONFIG)
+
+    # WHEN uploading a sample with the api
+    api.upload(
+        sample_id=sample_id,
+        sample_name=sample_name,
+        group_id=group_id,
+        group_name=group_name,
+        bed_file=bed_file,
+    )
+
+    # THEN run_command should be called with the correct arguments
+    mocked_run_command.assert_called_once_with(
+        [
+            "load",
+            "--sample",
+            sample_id,
+            "--name",
+            sample_name,
+            "--group",
+            group_id,
+            "--group-name",
+            group_name,
+            "--threshold",
+            "10",
+            str(bed_file),
+        ]
+    )
+
+
+def test_chanjo_api_sample_existing(mocker, mock_process):
+    """Test sample method"""
+
+    # GIVEN a mocked process with a mocked stdout and a chanjo api
+    sample_id = "sample_id"
+    mocked_stdout = '[{"id": "%s"}]' % sample_id
+    mocked_stderr = ""
+    MockedProcess = mock_process(
+        result_stderr=mocked_stderr, result_stdout=mocked_stdout
+    )
+    mocked_process = mocker.patch("cg.apps.coverage.api.Process")
+    mocked_process.return_value = MockedProcess(
+        binary=CHANJO_CONFIG["chanjo"]["binary_path"],
+        config=CHANJO_CONFIG["chanjo"]["config_path"],
+    )
+    api = ChanjoAPI(CHANJO_CONFIG)
+
+    # WHEN fetching an existing sample from the api
+    sample = api.sample(sample_id=sample_id)
+
+    # THEN sample should be a dictionary with key id = sample_id
+    assert sample["id"] == sample_id
+
+
+def test_chanjo_api_sample_non_existing(mocker, mock_process):
+    """Test sample method"""
+
+    # GIVEN a mocked process with a mocked stdout and a chanjo api
+    sample_id = "sample_id"
+    mocked_stdout = '[{"id": "%s"}]' % sample_id
+    mocked_stderr = ""
+    MockedProcess = mock_process(
+        result_stderr=mocked_stderr, result_stdout=mocked_stdout
+    )
+    mocked_process = mocker.patch("cg.apps.coverage.api.Process")
+    mocked_process.return_value = MockedProcess(
+        binary=CHANJO_CONFIG["chanjo"]["binary_path"],
+        config=CHANJO_CONFIG["chanjo"]["config_path"],
+    )
+    api = ChanjoAPI(CHANJO_CONFIG)
+
+    # WHEN fetching a non existing sample from the api
+    sample = api.sample(sample_id="non_existing")
+
+    # THEN None should have been returned
+    assert sample is None
+
+
+def test_chanjo_api_delete_sample(mocker):
+    """Test delete method"""
+    # GIVEN a process with a mocked run_command method
+    sample_id = "sample_id"
+    mocked_run_command = mocker.patch.object(Process, "run_command")
+    api = ChanjoAPI(CHANJO_CONFIG)
+
+    # WHEN deleting a sample with the api
+    api.delete_sample(sample_id=sample_id)
+
+    # THEN run_command should be called with the correct arguments
```

Comment:
I guess "correct" could be considered a bit vague? I think the intention of any THEN clause is to give some information about the system under test and and I'm uncertain what to learn from "correct"? 

I suggest we specify what "correct" is so the test can be used to rectify the implementation when it deviates from the business rule, right?

Reply:
I get your point. Would it be enough to specify in the THEN-comment what the 'correct' arguments for the call should be?